src/main/java/org/myrobotlab/service/config/ServiceConfig.java

```diff
@@ -1,15 +1,27 @@
 package org.myrobotlab.service.config;
 
 /**
- * Base service configuration class.  
- * All services must have a type.
- * The name of the service config file implies the 
- * name of the service.
+ * Base service configuration class. All services must have a type. The name of
+ * the service config file implies the name of the service.
  *
  */
 public class ServiceConfig {
 
-  // The type of the service defined in this config
+  /**
+   * type of service defined for this config
+   */
   public String type;
 
+  public ServiceConfig() {
+    String configTypeName = this.getClass().getSimpleName();
+    String serviceType = configTypeName.substring(0, configTypeName.length() - "Config".length());
```

Comment:
This means no services can live outside the org.myrobotlab.service package.. yes?    why not make type and name final?  force them to be in the constructor?

Reply:
Ignore that previous comment here.... 
I still think these should be made final and passed in.  I don't like having code that does stuff in a pojo.. especially error prone operations such as a substring.